### PR TITLE
IGNITE-11923 add IgniteMXBean attributes to metrics

### PR DESCRIPTION
**Pull request by zkillingjar**

### CODE REVIEW

File: modules/core/src/main/java/org/apache/ignite/mxbean/IgniteMXBean.java [hunk 1]
```diff
@@ -27,36 +27,152 @@
...
+    public String ATTR_DESC_READ_ONLY_MODE = "Cluster read-only mode status.";
+
+    /** */
+    public String ATTR_DESC_READ_ONLY_MODE_DURATION = "Duration of read-only mode enabled on cluster.";
+
     /**
      * Gets string presentation of the version.
      *
      * @return String presentation of the version.
+     * @deprecated Use metric API.
```
Comment (nizhikov) [inline modules/core/src/main/java/org/apache/ignite/mxbean/IgniteMXBean.java]: Let's replace this and all others deprecation comments with the comment that equal all over the code:\

` * @deprecated Use {@link GridMetricManager} instead.`

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/preloader/GridDhtPartitionDemander.java [hunk 2]
```diff
@@ -424,7 +424,7 @@ private void requestPartitions(final RebalanceFuture fut, GridDhtPreloaderAssign
             return;
         }
 
-        if (!ctx.kernalContext().grid().isRebalanceEnabled()) {
+        if (!ctx.isRebalanceEnabled()) {
```
Comment (nizhikov) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/preloader/GridDhtPartitionDemander.java]: Why this change?

File: modules/core/src/main/java/org/apache/ignite/internal/IgniteKernal.java [hunk 3]
```diff
@@ -596,14 +596,24 @@ public IgniteKernal(@Nullable GridSpringResourceContext rsrcCtx) {
 
     /** {@inheritDoc} */
     @Override public boolean isNodeInBaseline() {
-        ClusterNode locNode = localNode();
+        ctx.gateway().readLockAnyway();
```
Comment (nizhikov) [inline modules/core/src/main/java/org/apache/ignite/internal/IgniteKernal.java]: Why this change?

File: modules/core/src/test/java/org/apache/ignite/internal/metric/JmxExporterSpiTest.java [hunk 4]
```diff
@@ -730,6 +735,105 @@ public void testScanQuery() throws Exception {
         }
     }
 
+    /**
+     *
```
Comment (nizhikov) [inline modules/core/src/test/java/org/apache/ignite/internal/metric/JmxExporterSpiTest.java]: nit: Let's write this comment in one line.

File: modules/core/src/main/java/org/apache/ignite/mxbean/IgniteMXBean.java [hunk 5]
```diff
@@ -27,36 +27,152 @@
  */
 @MXBeanDescription("MBean that provides access to Kernal information.")
 public interface IgniteMXBean {
+    /** */
+    public String ATTR_DESC_FULL_VER = "String presentation of the Ignite version.";
```
Comment (nizhikov) [inline modules/core/src/main/java/org/apache/ignite/mxbean/IgniteMXBean.java]: Can we rename this constant to `FULL_VER_DESC`?
Other constants introduced by this change should be renamed with this pattern, also.

File: modules/core/src/test/java/org/apache/ignite/internal/metric/JmxExporterSpiTest.java [hunk 6]
```diff
@@ -730,6 +735,105 @@ public void testScanQuery() throws Exception {
...
+
+        int cnt = 36;
+
+        for (MBeanAttributeInfo info : mbn.getMBeanInfo().getAttributes()) {
+            cnt--;
+
+            assertFalse(F.isEmpty(info.getDescription()));
+            Object val = mbn.getAttribute(info.getName());
+
+            switch (info.getName()) {
```
Comment (nizhikov) [inline modules/core/src/test/java/org/apache/ignite/internal/metric/JmxExporterSpiTest.java]: Can we rewrite this test with the more dumb fashion? 

```
assertEquals(36, mbn.getMBeanInfo().getAttributes().size());

assertFalse(F.isEmpty(mbn.getAttribute("fullVersion")));
assertFalse(F.isEmpty(mbn.getAttribute("copyright")));
```

And so on...

File: modules/core/src/main/java/org/apache/ignite/mxbean/IgniteMXBean.java (hunk 1 above)
Comment (zkillingjar) [reply modules/core/src/main/java/org/apache/ignite/mxbean/IgniteMXBean.java]: done

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/preloader/GridDhtPartitionDemander.java (hunk 2 above)
Comment (zkillingjar) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/preloader/GridDhtPartitionDemander.java]: Removed it.

File: modules/core/src/main/java/org/apache/ignite/internal/IgniteKernal.java (hunk 3 above)
Comment (zkillingjar) [reply modules/core/src/main/java/org/apache/ignite/internal/IgniteKernal.java]: `localNode()` takes the gw lock internally and the latter throws an exception if state is not `STARTED`. As a result the test `ReadMetricsOnNodeStartUpTest` was failing.
Rechecked it now and it doesn't reproduce anymore. Most likely due to the fact that I moved the call to `registerMetrics()` past the place where gw state is set to `STARTED`.

So I reverted the change for isNodeBaselne() back to original.

File: modules/core/src/test/java/org/apache/ignite/internal/metric/JmxExporterSpiTest.java (hunk 4 above)
Comment (zkillingjar) [reply modules/core/src/test/java/org/apache/ignite/internal/metric/JmxExporterSpiTest.java]: done

File: modules/core/src/main/java/org/apache/ignite/mxbean/IgniteMXBean.java (hunk 5 above)
Comment (zkillingjar) [reply modules/core/src/main/java/org/apache/ignite/mxbean/IgniteMXBean.java]: done

File: modules/core/src/test/java/org/apache/ignite/internal/metric/JmxExporterSpiTest.java (hunk 6 above)
Comment (zkillingjar) [reply modules/core/src/test/java/org/apache/ignite/internal/metric/JmxExporterSpiTest.java]: done
